### ISSUE TITLE
Reimplement slimmer functionality

### ISSUE THREAD

**Issue by kltm** [geneontology/amigo#144]
There is still interest for online slimming functionality; the goal here is to first implement slimmer functionality with the improvements to the perl backend from #69, then move on to more interesting functionality (like defining your own cutsets using a tree browser).

**Comment by kltm** [comment]
From Val:
```
Will the web based GO slim tool be returning at any point?
see comments in this ticket
http://jira.geneontology.org/browse/GO-1027

PomBase currently point users to
http://go.princeton.edu/cgi-bin/GOTermMapper
rather than QuickGO because it matches our output
http://www.pombase.org/browse-curation/fission-yeast-go-slim-terms

We don't want to point our users to QuickGO because the GO annotations are differently in GOA than in GO
1. We have filtered all of the annotations generated by erroneous mappings submitted to this tracker
https://github.com/geneontology/go-annotation/issues
AND

2. GOA/QuickGO filter annotations for syntax errors which are currently accepted by the
GO loading script.
geneontology/go-site#123
```

**Comment by suzialeksander** [comment]
Bumping as there was recent interest in this from Helpdesk, spawned from the "Paris" virtual Users Meeting

**Comment by kltm** [comment]
@suzialeksander I think this is probably not going to happen in AmiGO. Once the functionality was originally removed, we briefly had it in Galaxy (remember that?) and that never really gained traction. The API covers some of this and this is possibly better tackled via web (Jupyter) notebook or the like.

**Comment by ValWood** [comment]
But slimmer functionality is not so far removed from the MAtrix tool  so why not provide it in the familiar AmiGO environment?

**Comment by kltm** [comment]
@ValWood At this point, and with many things that we'd like to do, limited resources and priorities. It may be possible to restart some of this when we start into AmiGO development again, but as it has been for the last few years (last note on ticket five years ago), we are unlikely to meet that need with our current roadmap with the current AmiGO.
That said, if there is a huge push for this, versus things like QC efforts or Noctua, then resources could be freed.

**Comment by ValWood** [comment]
OK. It does feel a little like the public facing and major products of GO  (ontology and web tools)  are a little deprioritised and starved of resources. But I guess it isn't possible to do *everything*

**Comment by lpalbou** [comment]
That's certainly something to discuss more. If we are to do more User Meetings, we'll have to improve our APIs and docs indeed.

As mentioned in the [other ticket](https://github.com/geneontology/geneontology.github.io/issues/251), creating a slim is a different task than mapping annotations to a set of higher level terms. The former, I don't see that happening any time soon but the later has been available for some time in the API and is used by the ribbon to create the association tables.

**Comment by ValWood** [comment]
> but the later has been available for some time in the API and is used by the ribbon to create the association tables.

This isn't something the average biologist could use to slim their own lists though.
They need a web interface .We cater far too much for informaticians and not enough to lab scientists.

The slim functionality at PomBase is *really *. popular. We don't have an API, and we've never been asked to provide one (although we should), but the point is most  of the GO (potential) user base would not know how to use the API, they will go elsewhere.

**Comment by lpalbou** [comment]
Agreed. I will try to reopen the discussion. Do you have a link to the pombase tool ? Thanks

**Comment by ValWood** [comment]
The slimmer functionality is embedded into our query tool.

https://www.pombase.org/query
You can upload any  pombe gene list and slim (now using process, function, component, or disease slim)

To try it you can select a list from the commonly used queries (all disease associated genes for example).

Click on the list in the results table.
One of the "actions" at the top right of the results table is to slim your list

very quick, very easy.

**Comment by ValWood** [comment]
The 'full' slim sets are available from the front page
https://www.pombase.org/

I don't have stats for the query tool, only for the static pages.
The new disease slim, relaunched last month using MONDO has been accessed 101 times by 63 unique users since May 1st. The BP slim has been accessed 88 times by 55 users.
I imagine many many more people use the slim feature  with their own queries.

Slims are really popular for people to orientate themselves with their data. For bench scientists, probably more so than enrichment tools if they are made easily avaiable, and the slim has good coverage. They can see immediately what they are dealing with and plan where to go next.
